### PR TITLE
Separate the Generator from World and Plugin interfaces

### PR DESCRIPTION
**Pull request by madmaxoft**
The generator now doesn't depend on any `cChunkSink` or `cPluginInterface`, its sole responsibility is to generate a single chunk on request.
Moved the generator defaults into the `cComposableGenerator`, rather than `cWorld`.
Added a basic test that checks the default Overworld and Nether generator.

### CODE REVIEW

File: src/ChunkGeneratorThread.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+
+
+
+
+void cChunkGeneratorThread::Execute(void)
+{
+	// To be able to display performance information, the generator counts the chunks generated.
+	// When the queue gets empty, the count is reset, so that waiting for the queue is not counted into the total time.
+	int NumChunksGenerated = 0;  // Number of chunks generated since the queue was last empty
```
Comment (NiLSPACE) [inline src/ChunkGeneratorThread.cpp]: Shouldn't variables start lower-cased?

File: src/ChunkGeneratorThread.h [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+		QueueItem(cChunkCoords a_Coords, bool a_ForceRegeneration, cChunkCoordCallback * a_Callback):
+			m_Coords(a_Coords),
+			m_ForceRegeneration(a_ForceRegeneration),
+			m_Callback(a_Callback)
+		{
+		}
+	};
+
+	using Queue = std::list<QueueItem>;
```
Comment (NiLSPACE) [inline src/ChunkGeneratorThread.h]: Wouldn't it make more sense to use a `std::queue` instead?

Comment (peterbell10) [reply src/ChunkGeneratorThread.h]: Or the queue from `OSSupport/Queue.h` which should be thread-safe already.

File: src/ChunkGeneratorThread.cpp (hunk 1 above)
Comment (madmaxoft) [reply src/ChunkGeneratorThread.cpp]: This is basically only code that has been moved, so I didn't change it.
Subconsciously I don't use the new conventions on old code just yet, I'll change everything at once when changing the API.

File: src/ChunkGeneratorThread.h (hunk 2 above)
Comment (madmaxoft) [reply src/ChunkGeneratorThread.h]: Again, code that has been just moved around.
`std::queue` is just a wrapper over `std::deque` by default.

OSSupport/Queue.h might be interesting, didn't remember we had one. I'll have a look at it.

Comment (madmaxoft) [reply src/ChunkGeneratorThread.h]: I has a look at the `cQueue`, but in the end, `cChunkGeneratorThread` does a bit more than just read and write to a queue, it calculates the throughput, and I'm too lazy to adapt that to use the `cQueue`, it isn't a straightforward change.

**Comment by madmaxoft** [comment]
Well if that isn't weird - Debug build tests succeed, Release build tests fail because the generated chunk doesn't seem to have any Nether-specific blocks. WAT?

**Comment by madmaxoft** [comment]
I'll hotfix the test for now (it doesn't consider nether quarts ore nor lava as a nether block), but we need to find out why there's differring data generated on debug / release builds, and whether it happens only on Linux, or on Windows, too.

**Pull request merged**